test/unit/lib/permalink_fu_test.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
```

Comment:
`test/unit/lib` is a Pandora's box. It's like having a `others` dir. I know it wasn't this PR the one to open the box, but please let's close it before it's too late. We have `app/lib` for convenience while moving things from `lib`, but even that one needs to be better organised IMO.